src/modules/Message/utils/tokens/__tests__/tokenizeUtils.spec.ts

```diff
@@ -64,15 +64,91 @@ describe('identifyUrlsAndStrings', () => {
     expect(result).toEqual([{
       type: 'string',
-      value: 'abc',
+      value: 'abc ',
     }, {
       type: 'url',
       value: 'https://www.google.com',
     }, {
       type: 'string',
-      value: '123',
+      value: ' 123',
+    }]);
+  });
+
+  it('should parse multiple urls with special charactors', () => {
+    const tokens = [{
+      type: 'undetermined',
+      value: '[https://www.google.com](https://www.google.com)',
+    }] as UndeterminedToken[];
+    const result = identifyUrlsAndStrings(tokens);
+    expect(result).toEqual([{
+      type: 'string',
+      value: '[',
+    }, {
+      type: 'url',
+      value: 'https://www.google.com',
+    }, {
+      type: 'string',
+      value: '](',
+    }, {
+      type: 'url',
+      value: 'https://www.google.com',
+    }, {
+      type: 'string',
+      value: ')',
     }]);
   });
+
+  it('should parse valid URLs correctly', () => {
+    const validURLs = [
+      // with protocol
+      'http://www.example.com',
+      'https://www.example.com',
+      'http://example.com',
+      'https://example.com',
+      // without protocol
+      'www.example.com',
+      'example.com',
+      // with sub paths
+      'http://www.example.com/path/to/page.html',
+      'https://www.example.com/path/to/page.html',
+      'http://example.com/path/to/page.html',
+      'https://example.com/path/to/page.html',
+      'www.example.com/path/to/page.html',
+      'example.com/path/to/page.html',
+      // with query strings
+      'http://www.example.com/path/to/page.html?query=string',
+      'https://www.example.com/path/to/page.html?query=string',
+      'http://example.com/path/to/page.html?query=string',
+      'https://example.com/path/to/page.html?query=string',
+      'www.example.com/path/to/page.html?query=string',
+      'example.com/path/to/page.html?query=string',
+      'https://www.amazon.com/Hacker-Playbook-Practical-Penetration-Testing/dp/1494932636/ref=sr_1_5?crid=1IKVPDXYF5NQG&keywords=hacker+guide&qid=1681333238&sprefix=hacker+guid%2Caps%2C148&sr=8-5',
+      // with the hash property
+      'https://example.com/path/to/page.html?query=string#hash',
+      'https://docs.google.com/document/d/19IccwdTIwNPJ_rGtsbi2Ft8dshaH4WiCXD5pder97VE/edit#heading=h.pve9ikkfqqzz',
+      // A subdomain has a hyphen
+      'https://send-bird.slack.com/archives/C065N4UQ77W/p1699931368643169?thread_ts=1699925671l395019&cid-Co65N4UQ77W',
+      // with long top-level domain
+      'https://send.bird.business/archives/C065N4UQ77W/p1699931368643169?thread_ts=1699925671l395019&cid-Co65N4UQ77W',
+    ];
+    validURLs.forEach((url) => {
+      const result = identifyUrlsAndStrings([
+        {
+          type: 'undetermined',
+          value: url,
+        },
+      ]);
+      expect(result).toEqual([
+        {
+          type: 'string',
+          value: '',
+        }, {
+          type: 'url',
+          value: url,
+        },
+      ]);
```

Comment:
```suggestion
      expect(result).toEqual([
        {
          type: 'url',
          value: url,
        },
      ]);
```

Reply:
Changes from here: https://github.com/sendbird/sendbird-uikit-react/pull/961/files#r1477733499

Reply:
Ahh yeah the default array value should be removed in the end https://github.com/sendbird/sendbird-uikit-react/pull/961/commits/d6d368f15db8836783c5f7906dbfb0191fda74e2.

Reply:
https://github.com/sendbird/sendbird-uikit-react/pull/961#discussion_r1477733499
Oh! I left comments about the details of this change in the above comment!